platform/bitcoin/api.go

```diff
@@ -89,6 +91,40 @@ func (p *Platform) getTxsByAddress(address string) ([]blockatlas.Tx, error) {
 	return txs, nil
 }
 
+func (p *Platform) CurrentBlockNumber() (int64, error) {
+	status, err := p.client.GetBlockNumber()
+	return status.Backend.Blocks, err
+}
+
+func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
+	block, err := p.client.GetTransactionsByBlock(num, 1)
+	if err != nil {
+		return nil, err
+	}
+	if block.Page < block.TotalPages {
+		var wg sync.WaitGroup
+		out := make(chan Block)
+		for i := int64(2); i <= block.TotalPages; i++ {
+			go p.client.GetTransactionsByBlockChan(num, i, out, &wg)
+		}
+
+		wg.Wait()
+		defer close(out)
+		for r := range out {
+			block.Transactions = append(block.Transactions, r.Transactions...)
```

Comment:
Can we normalize the all transactions here? Because we don't need to iterate the array again to normalize the transactions.

Reply:
we have tried, that will make code more tedious (one more loop)